cost/google/idle_vm_recommendations/google_vm_recommendations.pt

```diff
@@ -400,65 +397,81 @@ script "js_instance_cost_mapping", type:"javascript" do
+          if (recommendation['costUnits'] == null) {
+            recommendation['costUnits'] = 0
           }
-          if (recommendation.costNanos<0) {
-            recommendation.costNanos = recommendation.costNanos * -1
+          if (recommendation['costNanos'] == null) {
+            recommendation['costNanos'] = 0
           }
-          if (recommendation.costUnits<0) {
-            recommendation.costUnits = recommendation.costUnits * -1
+          if (ds_currency_reference[recommendation['currency']] !== undefined ) {
+          cur = ds_currency_reference[recommendation['currency']]['symbol']
+          } else {
+            cur = "$"
           }
-          strNano = "."+recommendation.costNanos
-          combine = recommendation.costUnits + parseFloat(strNano)
-          monthsDuration = parseFloat(recommendation.duration.replace("s",""))/2628288
-          monthlySavings = combine / monthsDuration
-          total = total+monthlySavings
-          vm['savings']= (Math.round( monthlySavings * 1000) / 1000)
           vm['savingsCurrency'] = cur
-          vm['priority'] = recommendation['priority']
-          vm['region'] = recommendations['region']
-          vm['primaryImpact'] = recommendation['primaryImpact']
-          vm['recommenderSubtype'] = recommendation['recommenderSubtype']
-          vm['state'] = recommendation['state']
-          vm['service'] = "Compute"
-          tags = []
-          if (vm['tags'] != null) {
-            if( typeof vm['tags'] == 'object') {
-              Object.keys(vm.tags).forEach(function(key) {
-                if (vm.tags[key] == null || vm.tags[key]=="") {
-                  vm.tags[key] = "null"
-                }
-                tags.push(String(key)+'='+String(vm.tags[key]))
-              });
-            } else {
-              tags.push(vm["tags"].replace(":","=null"))
-            }
+        } else {
+          cur = "$"
+          var separator = ","
+        }
+
+        //Convert Costs from Negative to Positive and add Units and Nanos
+        if (!(recommendation.costNanos == null || recommendation.costNanos == undefined)) {
+          if (recommendation.costNanos != 0) {
+            recommendation.costNanos = recommendation.costNanos * -1
           }
-          vm['tags'] = tags
-          instances.push(vm)
+        } else {
+          recommendation.costNanos = 0
         }
-      })
+
+        if (recommendation.costUnits<0) {
+          recommendation.costUnits = recommendation.costUnits * -1
+        }
+        var cost_nano = recommendation.costNanos * Math.pow(10, -9)
+        var combine = Number(recommendation.costUnits) + Number(parseFloat(cost_nano))
+        monthsDuration = parseFloat(recommendation.duration.replace("s",""))/2628288
+        monthlySavings = combine / monthsDuration
+        total = total+monthlySavings
+        vm['savings']= (Math.round( monthlySavings * 1000) / 1000)
+        vm['savingsCurrency'] = cur
```

Comment:
We have a 'description' field in the policy incident, however it is likely that this will return empty in the incident. e.g.,
![image](https://github.com/flexera-public/policy_templates/assets/92175447/d13ca45b-02cb-4394-ba22-b1b4f759d6b6)

Therefore we should add this line:
```suggestion
        vm['savingsCurrency'] = cur
        vm['description'] = recommendation['description']
```

Reply:
@nia-vf1 thank you! I applied this change.